scripts/check-queries.lua

```diff
@@ -31,18 +37,23 @@ local function do_check()
   for _, lang in pairs(parsers) do
     for _, query_type in pairs(query_types) do
       print('Checking '..lang..' '..query_type)
-      local ok, query = pcall(queries.get_query,lang, query_type)
+      local ok, query = pcall(queries.get_query, lang, query_type)
       if not ok then
         vim.api.nvim_err_writeln(query)
         last_error = query
       else
         if query then
           for _, capture in ipairs(query.captures) do
-            if not vim.startswith(capture, "_") -- We ignore things like _helper
-              and captures[query_type]
-              and not capture:find("^[A-Z]") -- Highlight groups
-              and not vim.tbl_contains(captures[query_type], capture) then
-              error(string.format("Invalid capture @%s in %s for %s.", capture, query_type, lang))
+            local is_valid = (
+              capture:find("^[A-Z]") and query_type == 'highlights'  -- Highlight groups.
+              or vim.startswith(capture, "_")  -- Helpers.
+              or not captures[query_type]  -- Skip undocumented query types.
+              or vim.tbl_contains(captures[query_type], capture)
+            )
+            if not is_valid then
+              local error = string.format("(x) Invalid capture @%s in %s for %s.", capture, query_type, lang)
+              print(error)
+              last_error = error
```

Comment:
We shouldn't add explicit mapping  with uppercase letters. Also we didn't want undocumented queries. There could be an exception for injections

Reply:
I'm not sure to understand what you mean.

> We shouldn't add explicit mapping with uppercase letters

we should fail if there are uppercase captures?

> Also we didn't want undocumented queries. There could be an exception for injections

They are documented

Reply:
> we should fail if there are uppercase captures?

yes

> They are documented

but you are about to remove the check that checks whether captures they are documented.

Reply:
> but you are about to remove the check that checks whether captures they are documented.

Do you mean this line

```
or not captures[query_type]  -- Skip undocumented query types.
```

that currently fails for indents, I can document them (or at least list them bc I'm not that familiar with the indent queries) if you want me to remove that check.

Reply:
Would be great to have that documented, if we make an effort into documenting the queries, let's make it all at once.

@kyazdani42 we made need your help in order to document the indent queries.